From 75d9106342646ccad9080f9852a1ca22014ef910 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Thu, 29 Nov 2012 08:55:12 +0000
Subject: [PATCH 02/53] ENGR00235626 FEC: Enable phy pause frame feature

Since some ethernet MAC flow control is relied on phy pause
status, both link partners exchange information via auto neg to
determine if both parties are capable of flow control.
Advertise phy pause frame to avoid pause frame is not responsed
by the other net node.

Signed-off-by: Fugang Duan  <B38611@freescale.com>
---
 drivers/net/fec.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/fec.c b/drivers/net/fec.c
index 2d83d54..3a4ca0c 100755
--- a/drivers/net/fec.c
+++ b/drivers/net/fec.c
@@ -1046,6 +1046,9 @@ static int fec_enet_mii_probe(struct net_device *ndev)
 	else
 		phy_dev->supported &= PHY_BASIC_FEATURES;
 
+	/* enable phy pause frame for any platform */
+	phy_dev->supported |= ADVERTISED_Pause;
+
 	phy_dev->advertising = phy_dev->supported;
 
 	fep->phy_dev = phy_dev;
-- 
1.7.10.4

